client/client-multi/client-c/src/kaa/platform-impl/common/kaa_tcp_channel.c

```diff
@@ -392,8 +390,9 @@ kaa_error_t kaa_tcp_channel_destroy_context(void *context)
     kaa_error_t error_code = KAA_ERR_NONE;
 
     if (channel->parser) {
-        if(channel->parser->payload)
+        if (channel->parser->payload) {
```

Comment:
You don't need to check argument of `free()`:
`man 3 free`
>  The free() function frees the memory space pointed to by ptr, which must have been returned by a previous call to malloc(), calloc(), or realloc().  Otherwise, or if free(ptr)  has  already  been  called
       before, undefined behavior occurs.  __If ptr is NULL, no operation is performed.__